### PR TITLE
Store package and channel data in postgres

### PR DESCRIPTION
**Pull request by mwrock**
This replaces the redis based package store and in-memory based channel store with a sharded postgres store. The migration is through the whole stack (web api and data layer). Many of the structures and messages that were previous used as web data views have been moved from deposrv to originsrv. deposrv has been removed as well as all `data_store` structures in `builder-depot`.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-depot/src/doctor.rs [hunk 1]
```diff
@@ -154,7 +153,6 @@ impl<'a> Doctor<'a> {
 
     fn run(mut self) -> Result<Report> {
         try!(self.init_fs());
-        try!(self.truncate_datastore(&self.depot.datastore));
```
Comment (chefsalim) [inline components/builder-depot/src/doctor.rs]: I believe this functionality is still needed to clear the DB (ignore if it got moved somewhere else).

File: components/builder-depot/src/doctor.rs [hunk 2]
```diff
@@ -192,9 +190,10 @@ impl<'a> Doctor<'a> {
             let mut archive = PackageArchive::new(PathBuf::from(entry.path()));
             match archive.ident() {
                 Ok(ident) => {
-                    match depotsrv::Package::from_archive(&mut archive) {
-                        Ok(object) => {
-                            try!(self.depot.datastore.packages.write(&object));
+                    match originsrv::OriginPackageCreate::from_archive(&mut archive) {
+                        Ok(package) => {
+                            let mut conn = Broker::connect().unwrap();
+                            conn.route::<originsrv::OriginPackageCreate, originsrv::OriginPackage>(&package).unwrap();
```
Comment (chefsalim) [inline components/builder-depot/src/doctor.rs]: May be good to the result if unwrap can fail

File: components/builder-depot/src/doctor.rs (hunk 1 above)
Comment (mwrock) [inline components/builder-depot/src/doctor.rs]: I didn't add that in because with the sharding it's more involved and I'm really not sure we even want to keep the doctor "as is". I'm thinking we change it to NOT truncate but to be able to scan a storage system and ADD packages that it does not already have. That is super simple to change and was thinking it would be a separate PR. Thoughts?

File: support/Procfile [hunk 3]
```diff
@@ -1,7 +1,6 @@
-redis: redis-server --dir tmp
 web: support/builder_web.sh
```
Comment (reset) [inline support/Procfile]: `support/builder_web.sh` doesn't have an execute bit so this procfile doesn't start up cleanly. Easy fix!

Comment (mwrock) [inline support/Procfile]: weird. I wonder why it works for me. I never `chmod`ed it

Comment (reset) [inline support/Procfile]: Weird! After I `chmod`'d it everything started up just fine!

Comment (mwrock) [inline support/Procfile]: anyhoo...added the bit

**Comment by reset** [comment]
Oh, and @mwrock, nobody can tell you that you're not a Rust programmer :)

**Comment by mwrock** [comment]
cool. @chefsalim and @reset any thoughts on the `doctor` comments above?

File: components/builder-depot/src/doctor.rs (hunk 1 above)
Comment (reset) [inline components/builder-depot/src/doctor.rs]: @mwrock `doctor` should be moved to the `originsrv` and be a separate binary that we output. I think it's completely alright for now to break it in anyway and just revisit it. The work that @nellshamrell is doing right now to migrate data from an older builder cluster to a new one will most likely be re-usable in places to fill in what a proper `repair`/`doctor` command would look like

**Comment by mwrock** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/222991084).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/222991084).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265
